lms/views/basic_lti_launch.py

```diff
@@ -40,6 +40,20 @@ def __init__(self, context, request):
         self.context.js_config.enable_lti_launch_mode()
         self.context.js_config.maybe_set_focused_user()
 
+    def basic_lti_launch(self, document_url=None, grading_supported=True):
+        """Do a basic LTI launch with the given document_url."""
+        self.sync_lti_data_to_h()
+        self.store_lti_data()
+        self.course_service.get_or_create(self.context.h_group.authority_provided_id)
+
+        if grading_supported:
+            self.context.js_config.maybe_enable_grading()
+
+        if document_url is not None:
+            self.context.js_config.add_document_url(document_url)
+
+        return {}
```

Comment:
Helper method added to deduplicate the various different LTI launch methods below